tests/src/test/scala/cats/tests/OptionTSuite.scala

```diff
@@ -285,10 +285,14 @@ class OptionTSuite extends CatsSuite {
     }
   }
 
-  test("OptionT.whenF[F, A] consistent with the same implementation of Option.when") {
-    val when = (c: Boolean, j: Int) => if (c) Some(j) else None
-    forAll { (li: List[Int], b: Boolean) =>
-      OptionT.whenF(b)(li).value should ===(li.map(when(b, _)))
+  test("OptionT.whenF[F, A] consistent with Option.when") {
+    def when[A]: (Boolean, A) => Option[A] = (c: Boolean, a: A) => if (c) Some(a) else None
+    forAll { (i: Int, b: Boolean) =>
+      OptionT.whenF[Id, Int](b)(i).value should ===(when(b, i))
+    }
+
+    forAll { (i: List[Int], b: Boolean) =>
+      OptionT.whenF[List, Int](b)(i).value should ===(when(b, i).sequence)
```

Comment:
I'm unsure whether I'd ever reason about code by this property, but I noticed it when I thought about the types and it seems to pass.  I'm curious whether others think this one is actually useful.

Reply:
The test case is very interesting to me, but certainly less obvious that it universally holds (not only in the case when F = List). 

Added abd798b to remove it for now.
:+1: to restore it if someone knows a good explanation about it.